tools/zbfind

```diff
@@ -961,17 +981,23 @@ class ZBPoller(Thread):
             if fc_frametype == DOT154_FCF_TYPE_MACCMD and packet[2] == "\xff\xff" and packet[3] == "\xff\xff":
                 continue
 
+
             # devid is a UID for the device Source Address and SPAN combination, concatenated
             # If this is a beacon frame, we use the sa and spanid.  If this is anything else (excluding ACK),
             # we use the sa and dpanid.  The dpanid of other frames should match the spanid of beacon frames.
-            if fc_frametype == DOT154_FCF_TYPE_BEACON: 
+            if fc_frametype == DOT154_FCF_TYPE_BEACON:
                 devid = "%02x%02x%02x%02x" % (ord(packet[4][0]), ord(packet[4][1]),
                         ord(packet[5][0]), ord(packet[5][1]))
                 dstpan = "0x%02x%02x" % (ord(packet[4][0]), ord(packet[4][1]))
 
             else:
-                devid = "%02x%02x%02x%02x" % (ord(packet[2][0]), ord(packet[2][1]),
-                        ord(packet[5][0]), ord(packet[5][1]))
+                if packet[5] != "":
+                    devid = "%02x%02x%02x%02x" % (ord(packet[2][0]), ord(packet[2][1]),
+                            ord(packet[5][0]), ord(packet[5][1]))
+                else:
+                    devid = "%02x%02x%02x%02x" % (ord(packet[2][0]), ord(packet[2][1]),
+                            ord('A'), ord('B'))
```

Comment:
@Crypt0s  - Can you help me understand this? Was reading this again trying to finally merge this and again didn't know why this would be...